docs/reference/api/platform/ATCmdParser.md

```diff
@@ -1,11 +1,37 @@
 ## ATCmdParser
 
-[Add description here]
+ATCmdParser is an Mbed OS compatible AT command parser. 
+
+It sends and receives AT commands to and from a communication device implementing the [FileHandle](./FileHandle.md) interface. It also implements AT command parsing, which validates the data format and separate command and data portion of AT transactions. For example, the UARTSerial communication driver implements the FileHandle interface, and you can use it with ATCmdParser to send and receive AT commands. The actual format of AT commands used depends on the communication device used.
```

Comment:
Note to self: The link will probably be https://os.mbed.com/docs/v5.6/reference/filehandle.html once we create and merge that file.